src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Components.Routing;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Resolves components for an application.
+    /// </summary>
+    internal static class RouteTableFactory
+    {
+        private static readonly string ComponentAssemblyName = typeof(IComponent).Assembly.FullName;
+        private static readonly ConcurrentDictionary<Assembly, RouteTable> Cache =
+            new ConcurrentDictionary<Assembly, RouteTable>();
+        public static readonly IComparer<RouteEntry> RoutePrecedence = Comparer<RouteEntry>.Create(RouteComparison);
+
+        public static RouteTable Create(Assembly appAssembly)
```

Comment:
`RouteTable` and it's components don't have any state on them, so it's safe to statically cache this. This avoids a bunch of template parsing that was previously done per-Router instance